merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/LocalConstraintService.java

```diff
@@ -0,0 +1,41 @@
+package gov.nasa.jpl.aerie.merlin.server.services;
+
+import gov.nasa.jpl.aerie.constraints.model.Violation;
+import gov.nasa.jpl.aerie.merlin.server.models.Constraint;
+import gov.nasa.jpl.aerie.merlin.server.models.PlanId;
+import gov.nasa.jpl.aerie.merlin.server.models.SimulationDatasetId;
+import gov.nasa.jpl.aerie.merlin.server.remotes.ConstraintRepository;
+import gov.nasa.jpl.aerie.merlin.server.remotes.postgres.ConstraintRunRecord;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class LocalConstraintService implements ConstraintService {
+  private final ConstraintRepository constraintRepository;
+
+  public LocalConstraintService(
+    final ConstraintRepository constraintRepository
+  ) {
+    this.constraintRepository = constraintRepository;
+  }
+
+  @Override
+  public void createConstraintRuns(final Map<Long, Constraint> constraintMap, final Map<Long, Violation> violations, final PlanId planId, final SimulationDatasetId simulationDatasetId) {
+    this.constraintRepository.insertConstraintRuns(constraintMap, violations, planId, simulationDatasetId.id());
+  }
+
+  @Override
+  public Map<Long, ConstraintRunRecord> getPreviouslyResolvedConstraints(List<Constraint> constraints) {
+    final var resolvedConstraintRuns = new HashMap<Long, ConstraintRunRecord>();
+    final var constraintIds = constraints.stream().map(Constraint::id).collect(Collectors.toList());
+    final var constraintRuns = constraintRepository.getSuccessfulConstraintRuns(constraintIds);
```

Comment:
I was trying to limit any logic other than pulling from the db from being in the repository code, if it makes more sense to map it there rather than inside the service I can move that code and change return types.

Reply:
I think it would make more sense to map it there, yeah.